com.reprezen.swagedit/src/com/reprezen/swagedit/validation/Validator.java

```diff
@@ -101,50 +97,65 @@
 	}
 
 	/*
-	 * Finds all duplicate keys in all objects present in the yaml document.
+	 * Finds all duplicate keys in all objects present in the YAML document.
 	 */
-	protected Set<SwaggerError> checkDuplicateKeys(Node node) {
-		Set<SwaggerError> errors = Sets.newHashSet();
-		
-		if (node.getNodeId() == NodeId.mapping) {
-			MappingNode n = (MappingNode) node;
+	protected Set<SwaggerError> checkDuplicateKeys(Node document) {
+		HashMultimap<Pair<Node, String>, Node> acc = 
+				HashMultimap.<Pair<Node, String>, Node>create();
 
-			Map<String, Set<Node>> duplicates = Maps.newHashMap();
-			for (NodeTuple tuple: n.getValue()) {
-				Node keyNode = tuple.getKeyNode();
+		collectDuplicates(document, acc);
 
-				if (keyNode.getNodeId() == NodeId.scalar) {
-					String key = ((ScalarNode) keyNode).getValue();
-					if (duplicates.containsKey(key)) {
-						duplicates.get(key).add(keyNode);
-					} else {
-						duplicates.put(key, Sets.newHashSet(keyNode));
-					}
-				}
+		Set<SwaggerError> errors = Sets.newHashSet();
+		for (Pair<Node, String> key: acc.keys()) {
+			Set<Node> duplicates = acc.get(key);
 
-				errors.addAll(checkDuplicateKeys(tuple.getValueNode()));
+			if (duplicates.size() > 1) {
+				for (Node duplicate : duplicates) {
+					errors.add(createDuplicateError(key.getValue(), duplicate));
+				}
 			}
+		}
+
+		return errors;
+	}
 
-			for (String key: duplicates.keySet()) {
-				Set<Node> values = duplicates.get(key);
-				if (values.size() > 1) {
-					for (Node duplicate: values) {
-						errors.add(new SwaggerError(
-								duplicate.getStartMark().getLine() + 1, 
-								IMarker.SEVERITY_WARNING, 
-								String.format(Messages.error_duplicate_keys, key)));
-					}
+	/*
+	 * This method iterates through the YAML tree to collect the pairs of Node x String 
+	 * representing an object and one of it's keys. Each pair is associated to a Set of 
+	 * Nodes which contains all nodes being a key to the pair's Node and having for value 
+	 * the pair's key.
+	 * Once the iteration is done, the resulting map should be traversed. Each pair having 
+	 * more than one element in its associated Set are duplicate keys. 
+	 */
```

Comment:
This is a comment, not a Java doc. I wonder if it was your intention.
Similar with other methods in this class.